ui/src/components/SessionProvider.tsx

```diff
@@ -27,8 +27,32 @@ export default function SessionProvider({
   const [session, setSession, clearSession] = useLocalStorage('session', null);
 
   useEffect(() => {
+    const clearSessionIfNecessary = async () => {
+      const config = await getConfig();
+      if (session && session.required === false) {
+        let authEnabled = false;
+        if (
+          config.authentication.methods.oidc.enabled ||
+          config.authentication.methods.github.enabled ||
+          config.authentication.methods.jwt.enabled
```

Comment:
im not sure if we need to check if they enabled a auth method without it being required right? i would think we only care if it goes from `auth.requred == false` to `auth.required == true` (and maybe the opposite as well?)

Reply:
@markphelps Yeah I struggled with this. What does enabled but not required mean for the browser?

Reply:
nothing. if its not required then the API wont enforce it and the browser shouldn't care